executor/stale_txn_test.go

```diff
@@ -996,3 +998,96 @@ func (s *testStaleTxnSerialSuite) TestStaleReadPrepare(c *C) {
 	tk.MustExec(fmt.Sprintf(`set transaction read only as of timestamp '%s'`, time1.Format("2006-1-2 15:04:05.000")))
 	c.Assert("execute p1", NotNil)
 }
+
+func (s *testStaleTxnSuite) TestStmtCtxStaleFlag(c *C) {
```

Comment:
If we forbid it in the plan, maybe there are two situations:
The first: 
```
select /*+ read_from_storage(tiflash[table_name]) */ ... from table_name as of timestamp xxx;
```
Does it expect to report errors? 

The second:
If the origin plan is use tiflash, does use as of will change to tikv?
```
explain analyze select count(*) from test.t;
+--------------------------+---------+---------+--------------+---------------+----------------------------------------------------------------------+--------------------------------+-----------+------+
| id                       | estRows | actRows | task         | access object | execution info                                                       | operator info                  | memory    | disk |
+--------------------------+---------+---------+--------------+---------------+----------------------------------------------------------------------+--------------------------------+-----------+------+
| StreamAgg_9              | 1.00    | 1       | root         |               | time:83.8372ms, loops:2                                              | funcs:count(1)->Column#4       | 372 Bytes | N/A  |
| └─TableReader_17         | 1.00    | 1       | root         |               | time:83.7776ms, loops:2, rpc num: 1, rpc time:83.5701ms, proc keys:0 | data:TableFullScan_16          | 152 Bytes | N/A  |
|   └─TableFullScan_16     | 1.00    | 1       | cop[tiflash] | table:t       | time:43ms, loops:1                                                   | keep order:false, stats:pseudo | N/A       | N/A  |
+--------------------------+---------+---------+--------------+---------------+----------------------------------------------------------------------+--------------------------------+-----------+------+
// How about below?
explain analyze select count(*) from test.t as of timestamp
```

Reply:
I think the current implementation will prevent tiflash being picked from stale read, but I'm not sure what will happen if we bind hint to use tiflash forcely

Reply:
Hint is just a hint. And it should be ignored:

> An incorrect or incomplete hint will not result in a statement error. -- PingCAP Docs

The current behavior is it will be ignored.